infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -113,28 +191,105 @@ public long getLastSnapSeqNum() {
         return query(null, LogReplicationMetadataType.LAST_SNAPSHOT_SEQ_NUM);
     }
 
+    public long getLastSnapAppliedSeqNum() {
+        return query(null, LogReplicationMetadataType.LAST_SNAPSHOT_APPLIED_SEQ_NUM);
+    }
+
     public long getLastProcessedLogTimestamp() {
         return query(null, LogReplicationMetadataType.LAST_LOG_PROCESSED);
     }
 
-    public void appendUpdate(TxBuilder txBuilder, LogReplicationMetadataType key, long val) {
+    /**
+     * Given a specific timestamp, get the most recent full snapshot sync's timestamp that has complete the
+     * transferred phase.
```

Comment:
Given an timestamp, it will provide a consistent view of the metadata. When we refactor the metadata structure, this will not be needed as one access of the DB to get all the metadata.